platform/lib/platform/material/media.ex

```diff
@@ -163,6 +165,32 @@ defmodule Platform.Material.Media do
     end)
   end
 
+  defp populate_geolocation(changeset) do
```

Comment:
It's important not to repeat code. Let's instead use existing code paths like `update_from_virtual_data` in `attribute.ex`. You can use the existing `Attribute.changeset(...)` function.